File: src/main/java/uk/co/eelpieconsulting/feedlistener/model/LatLong.java
package uk.co.eelpieconsulting.feedlistener.model;

import dev.morphia.annotations.Embedded;

@Embedded
public class LatLong {

    private double latitude;
    private double longitude;

    public LatLong() {
    }

    public LatLong(double latitude, double longitude) {
        if (latitude < -90 || latitude > 90) {
            throw new IllegalArgumentException("Latitude is out of bounds");
        }
        if (longitude < -180 || longitude > 180) {
            throw new IllegalArgumentException("Longitude is out of bounds");
        }
        this.latitude = latitude;
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    @Override
    public String toString() {
        return "LatLong [latitude=" + latitude + ", longitude=" + longitude + "]";
    }

    @Override
    public int hashCode() {
        final int prime = 31;
        int result = 1;
        long temp;
        temp = Double.doubleToLongBits(latitude);
        result = prime * result + (int) (temp ^ (temp >>> 32));
        temp = Double.doubleToLongBits(longitude);
        result = prime * result + (int) (temp ^ (temp >>> 32));
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return false;
        LatLong other = (LatLong) obj;
        if (Double.doubleToLongBits(latitude) != Double
                .doubleToLongBits(other.latitude))
            return false;
        if (Double.doubleToLongBits(longitude) != Double
                .doubleToLongBits(other.longitude))
            return false;
        return true;
    }

}

File: src/main/kotlin/uk/co/eelpieconsulting/feedlistener/model/LatLong.kt
package uk.co.eelpieconsulting.feedlistener.model

import dev.morphia.annotations.Entity

@Entity
class LatLong(val latitude: Double, val longitude: Double) {

    init {
        require(!(latitude < -90 || latitude > 90))
        { "Latitude is out of bounds" }
        require(!(longitude < -180 || longitude > 180))
        { "Longitude is out of bounds" }
    }

    override fun toString(): String {
        return "LatLong [latitude=$latitude, longitude=$longitude]"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as LatLong

        if (latitude != other.latitude) return false
        return longitude == other.longitude
    }

    override fun hashCode(): Int {
        var result = latitude.hashCode()
        result = 31 * result + longitude.hashCode()
        return result
    }

}